buildSrc/src/main/resources/checkstyle.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+  <property name="charset" value="UTF-8" />
+
+  <module name="TreeWalker">
+    <!-- ~3500 violations
+    <module name="LineLength">
+      <property name="max" value="140"/>
+    </module>
+    -->
+
+    <module name="AvoidStarImport" />
+    <!-- Doesn't pass but we could make it pass pretty quick.
+    <module name="UnusedImports">
+      The next property is optional. If we remove it then imports that are
+      only referenced by Javadoc cause the check to fail.
+      <property name="processJavadoc" value="true" />
+    </module>
+    -->
+
+    <!-- Non-inner classes must be in files that match their names. -->
+    <module name="OuterTypeFilename" />
+    <!-- No line wraps inside of import and package statements. -->
+    <module name="NoLineWrap" />
+    <!-- Each java file has only one outer class -->
+    <module name="OneTopLevelClass" />
+
+    <!-- We don't use Java's builtin serialization and we suppress all warning
+      about it. The flip side of that coin is that we shouldn't _try_ to use
+      it. We can't outright ban it with ForbiddenApis because it complain about
+      every we reference a class that implements Serializable like String or
+      Exception.
+      -->
+    <module name="RegexpSinglelineJava">
+      <property name="format" value="serialVersionUID" />
+      <property name="message" value="Do not declare serialVersionUID." />
+      <property name="ignoreComments" value="true" />
```

Comment:
That one I ported from forbidden patterns. We had plenty of violations and
I fixed them. Since they are only appropriate to java I moved them here
rather than leaving them in forbidden patterns. Also, you are now allowed
to mention serialVersionUID in a comment so the suppression is less
Orwellian.
On Jan 30, 2016 9:57 AM, "Jason Tedor" <notifications@github.com> wrote:

> In buildSrc/src/main/resources/checkstyle.xml
> <https://github.com/elastic/elasticsearch/pull/16279#discussion_r51345865>
> :
>
> > +    <module name="OuterTypeFilename" />
> > +    <!-- No line wraps inside of import and package statements. -->
> > +    <module name="NoLineWrap" />
> > +    <!-- Each java file has only one outer class -->
> > +    <module name="OneTopLevelClass" />
> > +
> > +    <!-- We don't use Java's builtin serialization and we suppress all warning
> > +      about it. The flip side of that coin is that we shouldn't _try_ to use
> > +      it. We can't outright ban it with ForbiddenApis because it complain about
> > +      every we reference a class that implements Serializable like String or
> > +      Exception.
> > +      -->
> > +    <module name="RegexpSinglelineJava">
> > +      <property name="format" value="serialVersionUID" />
> > +      <property name="message" value="Do not declare serialVersionUID." />
> > +      <property name="ignoreComments" value="true" />
>
> I'm curious why you have this? I don't see any violations today, and I
> don't foresee any violations?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/16279/files#r51345865>.
>